src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -728,4 +733,21 @@ object ScalafixPlugin extends AutoPlugin {
 
   private val scalacOptionsToRelax =
     List("-Xfatal-warnings", "-Werror", "-Wconf.*").map(_.r.pattern)
+
+  /** Try load a setting key from sbt.Keys using reflection. This is usefull to
+    * deal with the binary compatibility of different versions of sbt.
+    *
+    * For instance the bspEnabled key has been introduced in sbt 1.4. Calling
+    * tryLoadSbtSetting("bspEnabled") return None on sbt 1.3 and
+    * Some(Keys.bspEnabled) on sbt 1.4 and greater.
+    */
+  private def tryLoadSbtSetting[T](key: String): Option[SettingKey[T]] = {
```

Comment:
this will be handy! I wonder though if wouldn't it be simpler to declare a local `bspEnabled` key with the same type as the upstream one?

Reply:
I have tried that:
```scala
        settingKey[Boolean]("bspEnabled") := false,
```
but it did not work.

Do you know if there is a working example somewhere?

Reply:
I mean the original `bspEnabled` setting is still true after that.

Reply:
> Do you know if there is a working example somewhere?

I had good results copying https://github.com/scalacenter/bloop/blob/v1.4.0/integrations/sbt-bloop/src/main/scala/bloop/integrations/sbt/SbtBloop.scala#L76-L77 in a private project for a somewhat similar use-case. It just worked in that case, but `bspEnabled` seems to be a non-regular key in the way it's consumed?

Reply:
Indeed your solution does work.

> bspEnabled seems to be a non-regular key in the way it's consumed?

I would say it is a regular key, but `Global / bspEnabled` is consumed very early by sbt server itself.